lib/ProgramOptions/IniFileParser.h

```diff
@@ -36,6 +36,9 @@ namespace options {
 class IniFileParser {
  public:
   explicit IniFileParser(ProgramOptions* options) : _options(options) {
+    // a regex for removing all trailing comments
+    _matchers.removeComments =
+        std::regex("[ \t]?#.*$", std::regex::nosubs | std::regex::ECMAScript);
```

Comment:
why remove zero or one whitespaces in front of the `#`, and not either none or all?

Are we sure that there are no cases where a `#` can legitimately occur in a config option that we do not want to break? e.g. in a password, jwt token, filenames, ...?